### PR TITLE
Purge products

### PR DESCRIPTION
**Pull request by dizcology**
samples for new features:
* delete all products in a product set
* delete all products not in any product sets

### CODE REVIEW

**Comment by dizcology** [comment]
DO NOT MERGE until updated to `operation.result(timeout=60)`.

File: vision/cloud-client/product_search/product_management.py [hunk 1]
```diff
@@ -181,6 +183,39 @@ def delete_product(project_id, location, product_id):
...
+    operation = client.purge_products(
+        parent=parent,
+        delete_orphan_products=True,
+        # The operation is irreversible and removes multiple products.
+        # The user is required to pass in force=True to actually perform the
+        # purge.
+        # If force is not set to True, the service raises an exception.
+        force=force)
+
+    # TODO: replace with operation.result(timeout=60)
```
Comment (munkhuushmgl) [inline vision/cloud-client/product_search/product_management.py]: Java there is already method for this purpose, java--> ` response.get(60, TimeUnit.SECONDS);` However, C# --> `operation.Result.PollUntilCompleted();` which doesn't contain timeout option.   Do we really have to include this 60 sec timeout for every language?

**Comment by davidcavazos** [comment]
Once the `operation.result(timeout=60)` is in place, LGTM

File: vision/cloud-client/product_search/product_in_product_set_management.py [hunk 2]
```diff
@@ -117,6 +119,45 @@ def remove_product_from_product_set(
...
+    operation = client.purge_products(
+        parent=parent,
+        product_set_purge_config=product_set_purge_config,
+        # The operation is irreversible and removes multiple products.
+        # The user is required to pass in force=True to actually perform the
+        # purge.
+        # If force is not set to True, the service raises an exception.
+        force=force)
+
+    # TODO: replace with operation.result(timeout=60)
```
Comment (beccasaurus) [inline vision/cloud-client/product_search/product_in_product_set_management.py]: Adding a GitHub PR comment for this TODO to track it ~ this comment will disappear once the TODO is fixed :D

File: vision/cloud-client/product_search/product_management.py (hunk 1 above)
Comment (dizcology) [reply vision/cloud-client/product_search/product_management.py]: specifying the timeout is not required, and does not need to be consistent across languages.  if possible each language should handle this in the language-specific conventional approach.

**Comment by steffnay** [comment]
LGTM

**Comment by dizcology** [comment]
sample tests passing

**Pull request merged**